Feature/barra pesquisa

ra247362 opened the pull request:
Projeto da barra de pesquisa básica.

Closes #17

ra247362 committed:
feat(search): add basic search bar

Adds a basic search bar to the top of every page. Allows user to search for companies by name. User will get results based on matching results (currently, companies whose name contains any part of the query terms).

ra247362 committed:
feat(search): clickable links

Search results are now clickable, taking user directly to company page.

ra247362 committed:
feat(search): testing and cleanup

Added various tests cases to ensure proper search results and behavior. Also cleaned up unnecessary files in search app.

Lozavival reviewed (changes requested):
No arquivo search/tests.py, revise os testes de perfect_query, narrow_query e broad_query, utilize a asserção [assertSequenceEqual](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertSequenceEqual) em vez de verificar a presença de cada elemento individualmente. Além disso, no teste de null_query, utilize a asserção [assertIn](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn) para verificar se a sequência "Realize uma pesquisa!" está presente no conteúdo da response.

Fora isso, os outros arquivos estão ok.

ra247362 committed:
chore(search): fix asserts in testing

Minor changes to testing, using proper asserts in each case for the search app test file.

Lozavival reviewed (approved):
Tudo ok, mudanças aprovadas.
